dask/bag/core.py

```diff
@@ -458,7 +455,7 @@ def distinct(self):
 
         return type(self)(merge(self.dask, dsk, dsk2), b, 1)
 
-    def reduction(self, perpartition, aggregate):
+    def reduction(self, perpartition, aggregate, split_every=None, out=Item):
```

Comment:
Might want to document `out`. Perhaps `out_type` as well, just to be a bit more descriptive? Otherwise, this looks good.